hoomd/ParticleData.cc

```diff
@@ -73,7 +73,7 @@ std::string getDefaultTypeName(unsigned int id)
     Type mappings assign particle types "A", "B", "C", ....
 */
 ParticleData::ParticleData(unsigned int N,
-                           const BoxDim& global_box,
+                           const std::shared_ptr<const BoxDim> global_box,
```

Comment:
It doesn't need it, but all the const modify should do in this case is prevent a change in what the smart pointer points to, I believe which is what we want since we are copying the box anyways.

Reply:
Yah agreed, it just reads weirdly for a `shared_ptr`, like if you used `const unsigned int N` rather than `unsigned int N` above, for example. It doesn't hurt anything, but it's not a pattern I'm used to seeing.

Reply:
I have inconsistently been striving to make as many things `const` as possible when editing the C++ code, but if you feel strongly, I can change it.

Reply:
I don't feel strongly, so whatever style you and @joaander want to enforce is fine by me! The reason I thought it looked unusual is that the raw pointer equivalent is `const BoxDim* const` (const pointer to const), and this is not really found elsewhere in HOOMD. The const qualifier could keep someone from accidentally trying to modify the pointer itself when passed by value, but this would be a big whoops (like trying to assign into `N`, for example).